Plutarch/Internal.hs

```diff
@@ -104,6 +104,15 @@ mapTerm f (TermResult t d) = TermResult (f t) d
 
 mkTermRes :: RawTerm -> TermResult
 mkTermRes r = TermResult r []
+
+-- | Type of `s`.
+data S
+
+-- | Shorthand for Plutarch types.
+type PType = S -> Type
+
+type role Term phantom representational
```

Comment:
I think the lack of this type role allowed `Term s PPubKeyHash` to be coerced to `Term s PByteString`. `coerce` no longer allows this. The change makes sense though - as it'd allow other unsafe conversions as well.

But I'm curious as to how one would do the above without `punsafeCoerce` now?

Reply:
We have `pto`, but `PlutusType` isn't implemented for most things, see #126.